src/Chapter3.hs

```diff
@@ -480,6 +514,49 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
 
+data City = City
+  { cityCastle       :: Castle
+  , cityMainBuilding :: Building
+  , cityHouses       :: [Building]
+  }
+
+data Castle
+  = WithoutCastle
+  | JustCastle
+  | SecureCastle
+
+data Building
+  = Church
+  | Library
+  | BuildingForOne
+  | BuildingForTwo
+  | BuildingForThree
+  | BuildingForFour
+
+peopleInBuilding :: Building -> Int
+peopleInBuilding building = case building of
+  BuildingForOne   -> 1
+  BuildingForTwo   -> 2
+  BuildingForThree -> 3
+  BuildingForFour  -> 4
```

Comment:
Unfortunately, as you don't pattern match on all the constructors, this function is partial – it will fail in runtime when you call for example `peopleInBuilding Church`, though it compiles okay, as types are fine.
The compiler would warn you that this pattern matching is not exhaustive 🙂